pkg/nsx/services/securitypolicy/firewall.go

```diff
@@ -528,20 +796,38 @@ func (service *SecurityPolicyService) createOrUpdateProjectGroups(obj *v1alpha1.
 
 func (service *SecurityPolicyService) ListSecurityPolicyID() sets.String {
 	// List SeurityPolicyID to which groups resources are associated in group store
-	groupSet := service.groupStore.ListIndexFuncValues(common.TagScopeSecurityPolicyCRUID)
+	groupSet := service.groupStoreForCR.ListIndexFuncValues(common.TagScopeSecurityPolicyCRUID)
 	// List SeurityPolicyID to which share resources are associated in share store
-	shareSet := service.shareStore.ListIndexFuncValues(common.TagScopeSecurityPolicyCRUID)
-	policySet := service.securityPolicyStore.ListIndexFuncValues(common.TagScopeSecurityPolicyCRUID)
+	shareSet := service.shareStoreForCR.ListIndexFuncValues(common.TagScopeSecurityPolicyCRUID)
+	policySet := service.securityPolicyStoreForCR.ListIndexFuncValues(common.TagScopeSecurityPolicyCRUID)
+
+	return groupSet.Union(policySet).Union(shareSet)
+}
+
+func (service *SecurityPolicyService) ListNetworkPolicyID() sets.String {
+	// List ListNetworkPolicyID to which groups resources are associated in group store
+	groupSet := service.groupStoreForNP.ListIndexFuncValues(common.TagScopeNetworkPolicyUID)
+	// List service to which share resources are associated in share store
+	shareSet := service.shareStoreForNP.ListIndexFuncValues(common.TagScopeNetworkPolicyUID)
+	policySet := service.securityPolicyStoreForNP.ListIndexFuncValues(common.TagScopeNetworkPolicyUID)
 
 	return groupSet.Union(policySet).Union(shareSet)
 }
 
 func (service *SecurityPolicyService) Cleanup() error {
 	// Delete all the security policies in store
 	uids := service.ListSecurityPolicyID()
-	log.Info("cleaning up security policies", "count", len(uids))
+	log.Info("cleaning up security policies created for CR", "count", len(uids))
+	for uid := range uids {
+		err := service.DeleteSecurityPolicy(types.UID(uid), true, common.ResourceTypeSecurityPolicy)
+		if err != nil {
+			return err
+		}
+	}
+	uids = service.ListNetworkPolicyID()
```

Comment:
Agreed, for non-vpc case, there is no SPs created for NP, we will not listed any SPs from NetworkPolicy UID, it should be ok to avoid unexpected cleanup actions.